tests/js/common/shell/shell-query-cache-noncluster.js

```diff
@@ -619,7 +619,7 @@ function AqlQueryCacheSuite () {
     
     testMaxResultsSize1 : function () {
       const query = "FOR i IN 1..@value RETURN i";
-      cache.properties({ mode: "demand", maxResults: 10, maxResultsSize: 8192 });
+      cache.properties({ mode: "demand", maxResults: 10, maxResultsSize: (1000 + 1001) * 9 + 100 * 2 });
```

Comment:
8192 ->18209 quite a huge size growth. As AqlValue sizeof was not changed, it is most likely caused by increased VPack size produced by `initFromUint` / `initFromInt`. Is this the only consequence?

Reply:
Yes, it's changed because vpack format changed, and I think it's good

Compression can be used to replace memory overhead via cpu overhead

Reply:
I understand that this PR removes the int48 type and we are now using int64/uint64 instead. Fine.
But the change should only be inside `AqlValue`. And both int48 and int64/uint64 can be stored inside an AqlValue and do not need extra memory. And all the numeric AqlValue types return `0` as their memory usage.
So I don't see why the reported memory usage should be higher than before now. What am I missing?

Reply:
@jsteemann  initFromUint / initFromInt  with this PR produce Slices with fixed format. e.g. even if number requires 1 byte  - 8 would be used. While it is inside AqlValue it makes no problem. But when this slice is copied anywhere from AqlValue it starts to overuse memory.

Reply:
Ah ok, makes sense.
The code in `AqlValue::toVelocyPack()` adds the inline `Slice` to the result `Builder` as is. It does not convert any numeric values into the shortest possible VelocyPack number type. We could easily add that optimization in `AqlValue::toVelocyPack(...)` to save memory.

Reply:
Something like the following should do it:
```diff
diff --git a/arangod/Aql/AqlValue.cpp b/arangod/Aql/AqlValue.cpp
index 0e7cff1ff14..a71069d299b 100644
--- a/arangod/Aql/AqlValue.cpp
+++ b/arangod/Aql/AqlValue.cpp
@@ -872,10 +872,13 @@ void AqlValue::toVelocyPack(VPackOptions const* options, VPackBuilder& builder,
         break;
       }
       [[fallthrough]];
-    case VPACK_INLINE:
-    case VPACK_INLINE_INT48:
     case VPACK_INLINE_INT64:
+      builder.add(VPackValue(_data.longNumberMeta.data.intLittleEndian.val));
+      break;
     case VPACK_INLINE_UINT64:
+      builder.add(VPackValue(_data.longNumberMeta.data.uintLittleEndian.val));
+      break;
+    case VPACK_INLINE:
     case VPACK_INLINE_DOUBLE:
     case VPACK_MANAGED_SLICE:
     case VPACK_MANAGED_STRING:
```

Reply:
note: above code is untested, but it should convey the idea.

Reply:
Do you want it?

Reply:
The code is missing conversion from little endian to host btw

Reply:
Ok, thanks for picking up the suggestion! :+1: